Plugins/Public/balancemagic/Main.cpp

```diff
@@ -198,62 +229,68 @@ bool UserCmd_Process(uint iClientID, const wstring &wscCmd)
-		{
-			float curr, max;
-			bool bShieldsUp;
-
-			if (subObjID == 1) // 1 is base (hull)
-				pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
-			else if (subObjID == 65521) // 65521 is shield (bubble, not equipment)
-				pub::SpaceObj::GetShieldHealth(iDmgToSpaceID, curr, max, bShieldsUp);
-			else
-				return; // If hit mounted equipment - do not continue with uninitialized variables.
+		return;
+	}
+	const auto& iter = mapDamageAdjust.find(iDmgMunitionID);
+	if (iter == mapDamageAdjust.end())
+	{
+		return;
+	}
 
-			uint iTargetType;
-			pub::SpaceObj::GetType(iDmgToSpaceID, iTargetType);
+	const DamageMultiplier& dmgInfo = iter->second;
+	float curr, max;
+	bool bShieldsUp;
+
+	if (subObjID == 1) // 1 is base (hull)
+		pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
+	else if (subObjID == 65521) // 65521 is shield (bubble, not equipment)
+		pub::SpaceObj::GetShieldHealth(iDmgToSpaceID, curr, max, bShieldsUp);
+	else if (subObjID < 34) // collision groups, external equipment starts at 34 onwards
+		curr = setHealth + (dmgInfo.projectileDamage / PLAYER_COLLISION_GROUP_HIT_PTS_SCALE);
+	else // external equipment (shield, thrusters, guns)
+		curr = setHealth + (dmgInfo.projectileDamage / PLAYER_ATTACHED_EQUIP_HIT_PTS_SCALE);
+
+	if (iDmgToSpaceID == 0) { // for external equipment, this value isn't populated and needs to be fetched
+		pub::Player::GetShip(iDmgTo, iDmgToSpaceID);
+	}
+	uint iTargetType;
+	pub::SpaceObj::GetType(iDmgToSpaceID, iTargetType);
 
-			// Deduce: if not fighter nor freighter, then it's obviously solar object.
-			if (iTargetType != OBJ_FIGHTER && iTargetType != OBJ_FREIGHTER)
-			{
-				setHealth = curr - (curr - setHealth) * iter->second.solar;
-			}
-			else
-			{
-				uint iArchID;
-				pub::SpaceObj::GetSolarArchetypeID(iDmgToSpaceID, iArchID);
-				uint targetShipClass = Archetype::GetShip(iArchID)->iShipClass;
-
-				if (targetShipClass == 0 || targetShipClass == 1 || targetShipClass == 3)
-					setHealth = curr - (curr - setHealth) * iter->second.fighter;
-				else if (targetShipClass == 2 || targetShipClass == 4 || targetShipClass == 5 || targetShipClass == 19)
-					setHealth = curr - (curr - setHealth) * iter->second.freighter;
-				else if (targetShipClass < 11)
-					setHealth = curr - (curr - setHealth) * iter->second.transport;
-				else if (targetShipClass < 13)
-					setHealth = curr - (curr - setHealth) * iter->second.gunboat;
-				else if (targetShipClass < 15)
-					setHealth = curr - (curr - setHealth) * iter->second.cruiser;
-				else if (targetShipClass < 16)
-					setHealth = curr - (curr - setHealth) * iter->second.battlecruiser;
-				else if (targetShipClass < 19)
-					setHealth = curr - (curr - setHealth) * iter->second.battleship;
-			}
+	// Deduce: if not fighter nor freighter, then it's obviously solar object.
+	if (iTargetType != OBJ_FIGHTER && iTargetType != OBJ_FREIGHTER)
+	{
+		setHealth = curr - (curr - setHealth) * dmgInfo.classMultipliers[SOLAR_INDEX];
```

Comment:
how about we just move the solar multiplier out into its own member property of DamageMultiplier?